src/scripts/app.ts

```diff
@@ -840,19 +841,23 @@ export class ComfyApp {
         // Assertion: Not yet defined in litegraph.
         const { newGraph } = e.detail
 
-        const nodeSet = new Set(newGraph.nodes)
+        const widgetIds: Record<string, BaseDOMWidget<object | string>> = {}
         const widgetStore = useDomWidgetStore()
 
-        // Assertions: UnwrapRef
-        for (const { widget } of widgetStore.activeWidgetStates) {
-          if (!nodeSet.has(widget.node)) {
-            widgetStore.deactivateWidget(widget.id)
-          }
-        }
+        for (const node of newGraph.nodes)
+          for (const w of node.widgets ?? [])
+            if (w instanceof DOMWidgetImpl && w.id) widgetIds[w.id] = w
 
-        for (const { widget } of widgetStore.inactiveWidgetStates) {
-          if (nodeSet.has(widget.node)) {
-            widgetStore.activateWidget(widget.id)
+        // Assertions: UnwrapRef
+        for (const widgetId of widgetStore.widgetStates.keys()) {
+          const widgetState = widgetStore.widgetStates.get(widgetId)
+          //Unreachable, but required for type safety
+          if (!widgetState) continue
+          if (widgetId in widgetIds) {
```

Comment:
This makes me think that `widgetIds` might be missing something in its name that helps describe its purpose.